### ISSUE TITLE
Extract the Hook concept that is not strictly related with WordPress and put it into a different repo

### ISSUE THREAD

**Issue by widoz** [widoz/wp-hooks#7]
There are part of the implementation which are not related to WordPress and can be reused even outside. 

Would be better to move them in another repo.

**Comment by szepeviktor** [comment]
Isn't it https://github.com/widoz/hooks-interfaces?

**Comment by widoz** [comment]
No, I noticed there are classes here which are not stricly related to WordPress and I thought to move them into another package. 

Other solution may then reuse those classes without the needs to have Wp stuffs as dependencies. 

The Hooks interfaces are a set of interfaces you can depends on without the real need to have implementations. eg, another library may consume the hooks but they don't need to have implementation just rely on the abstraction.

Basically I'm trying to serve only what's really needed no more moving different context into appropiate libs.

**Comment by widoz** [comment]
I created this repo https://github.com/widoz/hooks-collection to collect them

**Comment by widoz** [comment]
Depends on #24
